packages/fmg-nitro-adjudicator/contracts/NitroAdjudicator.sol

```diff
@@ -183,54 +210,62 @@ contract NitroAdjudicator {
             newDestination,
             allocation.finalizedAt,
             allocation.challengeCommitment,
-            zeroAddress,
             newAllocation
         );
     }
 
-    function affords(address recipient, Outcome memory outcome, uint funding) internal pure returns (uint256) {
+    function affords(
+        address recipient,
+        Outcome memory outcome,
+        uint funding
+    ) internal pure returns (uint256) {
         uint result = 0;
+        uint remainingFunding = funding;
 
         for (uint i = 0; i < outcome.destination.length; i++) {
-            if (funding <= 0) {
+            if (remainingFunding <= 0) {
                 break;
             }
 
             if (outcome.destination[i] == recipient) {
                 // It is technically allowed for a recipient to be listed in the
                 // outcome multiple times, so we must iterate through the entire
                 // array.
-                result =result.add(min(outcome.allocation[i], funding));
+                result = result.add(min(outcome.allocation[i], remainingFunding));
             }
-            if (funding > outcome.allocation[i]){
-                funding = funding.sub(outcome.allocation[i]);
+            if (remainingFunding > outcome.allocation[i]){
+                remainingFunding = remainingFunding.sub(outcome.allocation[i]);
             }else{
-                funding = 0;
+                remainingFunding = 0;
             }
         }
 
         return result;
     }
 
-    function reduce(Outcome memory outcome, address recipient, uint amount) internal pure returns (Outcome memory) { 
+    function reduce(
+        Outcome memory outcome,
+        address recipient,
+        uint amount
+    ) internal pure returns (Outcome memory) {
         uint256[] memory updatedAllocation = outcome.allocation;
         uint256 reduction = 0;
+        uint remainingAmount = amount;
```

Comment:
Is there a reason we don't want to modify the function parameter `amount`?

Reply:
The solidity linter complained about it. I didn't actually try to figure out why.